javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -360,6 +361,8 @@ final class Formatter private (private[this] var dest: Appendable,
             formatNumericString(localeInfo, flags, width, arg.toString())
           case arg: Long =>
             formatNumericString(localeInfo, flags, width, arg.toString())
+          case arg: BigInteger =>
```

Comment:
@sjrd is the optimizer able to remove this if `BigInteger` is unused?

Reply:
No, but it will only reach the instance tests. Therefore something like
```js
if ($is_jl_BigInteger(arg)) {
  result = arg.toString__T()
}
```
will be generated. But the call to `toString__T` will not actually reach `BigInteger.toString`, since there is no instance of `BigInteger`. Therefore, no additional code will be generated, beside those useless 3 lines.

Reply:
But do we even define `$is_jl_BigInteger`? `BigInteger` is a class, so IIUC, we'd just do that inline: `arg instanceof $c_jl_BigInteger`. And if the class isn't defined, the `Emitter` will fold this into `false`, no?

Reply:
Ah yes right, so we'll end up with
```js
if (false) {
  result = arg.toString__T()
}
```
(I think)

If the optimizer were taught how to get rid of impossible `IsInstanceOf` nodes like that, it would entirely eliminate the `if`.